### PR TITLE
Rename devpy to spin

### PR DESCRIPTION
**Pull request by jarrodmillman**
This is ready to review; spin 0.3 has been released:
https://pypi.org/project/spin/0.3/

### CODE REVIEW

File: spin
Comment (lagru) [inline spin:1]: Did you intentionally strip the `.py` extension. I think I'd prefer it as `spin.py`. That way tools globbing for `*.py` files are able to pick it up and Windows users can more easily see the file type at a glance.

Comment (stefanv) [reply spin:1]: Which tools would be globbing for *.py files? Adding `.py` just means more typing, so I don't think we'll make that the default recommendation, but skimage is free to change it to whatever.

Comment (lagru) [reply spin:1]: I disagree. Stripping `.py` comes at the cost of expressiveness and maybe general compatibility. The extension makes it easier to filter for Python files, e.g. IDE's, linting tools, other tooling or commands people use that we are not aware about. Is Windows still able to identify this as a Python file just based on the shebang? Why would we go this route if spin would not recommend it itself?

To me it just seems to be a dubious trade-off. Saving a bit of typing in a world where auto-complete, shell aliases, etc. exists is something I value less than expressiveness, compatibility and established expectations. There also isn't a precedent with similar tools that I know of, e.g. NumPy's runtests.py.

To be clear I won't block this, I just recommend against it. :wink:

**Comment by hmaarrfk** [comment]
PS. https://github.com/conda-forge/staged-recipes/pull/22319

File: spin
Comment (stefanv) [reply spin:1]: > I disagree. Stripping `.py` comes at the cost of expressiveness and maybe general compatibility. The extension makes it easier to filter for Python files, e.g. IDE's, linting tools, other tooling or commands people use that we are not aware about. Is Windows still able to identify this as a Python file just based on the shebang? Why would we go this route if spin would not recommend it itself?
> 
> To me it just seems to be a dubious trade-off. Saving a bit of typing in a world where auto-complete, shell aliases, etc. exists is something I value less than expressiveness, compatibility and established expectations. There also isn't a precedent with similar tools that I know of, e.g. NumPy's runtests.py.
> 
> To be clear I won't block this, I just recommend against it. 😉

I'm not sure why you say spin doesn't recommend it. It's supposed to, so if it doesn't it's an oversight.

The script will work fine on git bash style shells on Windows. I don't know about powershell, there `python -m spin` may be the preferred approach.

The concerns you raise are vague: which tools need to know that this is a Python file, and why? Again, if you don't like it, just rename it, I don't care what skimage uses as long as it's documented.

Comment (lagru) [reply spin:1]: > I don't think we'll make that the default recommendation

I understood this statement to mean that the spin project will not recommend to use the name `spin` but something else? But I see that it is now [recommending to use `spin`](https://github.com/scientific-python/spin/blob/5727c01752fa453172a31747c62f4bce42236e39/example_pkg/spin).

I'd rather say my arguments are abstract. But if you want a concrete example, e.g. after renaming `dev.py` to `spin` the command `ruff check .` (lint the current directory) doesn't pick up errors in `spin` anymore (`ruff check spin` and the commit hook still do).

<details>
<summary>Patch to demonstrate</summary>

```
Subject: [PATCH] Demo ruff not picking up the file
---
Index: dev.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/dev.py b/spin
rename from dev.py
rename to spin
--- a/dev.py	(revision 2ea79a6b4b309d8d73e10bb29ef3eb31981345e7)
+++ b/spin	(date 1679311296005)
@@ -7,6 +7,7 @@
 import os
 import sys
 import runpy
+import runpy
 
 sys.path.remove(os.path.abspath(os.path.dirname(sys.argv[0])))
 try:

```
</details>

> Again, if you don't like it, just rename it

That's not how I read you previous comment. Though I would prefer to reach a consensus on this so I'll open a PR on scientific-python/spin about this.

**Review by lagru — APPROVED** [review]
I updated a few remaining references to `dev.py`. Looks good now (except for the `.py` debate :wink:). Thanks!

**Comment by stefanv** [comment]
This PR brings us in line with the latest release of `spin`, so I'm going to merge for now. @lagru I hope the `spin` console script addresses your concerns around the `.py` extension, but let me know if not and we can discuss further.

**Comment by lagru** [comment]
No worries. I'm very happy with the "install `spin` as a console script" route. :)

**Pull request merged**